### PR TITLE
feat(text-editor): implement `readonly` prop

### PR DESCRIPTION
**Pull request by Kiarokh**
## Review:
- [ ] Commits are [atomic](https://seesparkbox.com/foundry/atomic_commits_with_git)
- [ ] Commits have the correct *type* for the changes made
- [ ] Commits with *breaking changes* are marked as such

### Browsers tested:
(Check any that applies, it's ok to leave boxes unchecked if testing something didn't seem relevant.)

Windows:
- [ ] Chrome
- [ ] Edge
- [ ] Firefox

Linux:
- [ ] Chrome
- [ ] Firefox

macOS:
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

Mobile:
- [ ] Chrome on Android
- [ ] iOS

### CODE REVIEW

File: src/components/text-editor/examples/text-editor-as-form-component.tsx [hunk 1]
```diff
@@ -1,20 +1,28 @@
...
     shadow: true,
 })
-export class TextEditorBasicExample {
+export class TextEditorAsFormComponentExample {
     @State()
     private formData: FormComponentFormData = {
-        name: 'Ali',
-        value: '<p>I am the greatest</p>',
+        name: 'Muhammad Ali',
+        value: '<p>I am the <b>greatest</b>.</p>',
```
Comment (Kiarokh) [inline src/components/text-editor/examples/text-editor-as-form-component.tsx:25]: @LawrenceBorst @civing not related to the current PR, but shouldn't we be able to give a markdown as `value`? When I remove the `<p>` tags, and replace the `<b>` tags with `*`, I will get the string straightly here. Or do we need to do something else?

File: src/components/text-editor/prosemirror-adapter/partial-styles/_prosemirror-menu.scss [hunk 2]
```diff
@@ -15,6 +15,14 @@ div#editor {
...
         }
     }
+
+    .ProseMirror-menubar-spacer {
+        // This is a div that they add to keep the menubar in place.
+        // as they are using `position: fixed`.
+        // However, this causes layout shift as the user scrolls,
+        // due to our usage of `position: sticky`.
+        display: none;
+    }
```
Comment (Kiarokh) [inline src/components/text-editor/prosemirror-adapter/partial-styles/_prosemirror-menu.scss:25]: https://github.com/Lundalogik/lime-elements/assets/35954987/97f9be09-fea7-42bd-89f0-9c4908a1f45c

File: src/components/text-editor/examples/text-editor-as-form-component.tsx (hunk 1 above)
Comment (LawrenceBorst) [reply src/components/text-editor/examples/text-editor-as-form-component.tsx:25]: Did you try with `I am the **greatest**.`? I tried it a while back

**Review by LawrenceBorst — APPROVED** [review]
Looks good to me

**Review by LawrenceBorst — APPROVED** [review]

**Pull request merged**